README
"make options:"

  clean   - remove all objects and executables"
  all     - build everything"
  tar     - create a release source tar using VERSION"

  sioseis - build just sioseis"
  sio2sun - build just sio2sun"
  sio2hp  - build just sio2hp"
  lsd     - build just lsd"
  lsh     - build just lsd"
  dutil   - build just dutil"

makefile uses FC=gfortran and CC=gcc.
Many compiler flags are commented out in an effort of finding what
compiles on "every" machine I could find.  The most common denominator.

As of 2010, tape I/O probably will not work on non Sun computers
    (mtio.h and other issues)

Make sure to use compatible versions of gcc and gfortran.
(At IGPP, there are two gccs, so the one used to compile gfortran must be used).


OSX:
X86_64 machines may have to use CFLAGS = -m32
CFLAGS = -O2 and other CFLAGS may be turned on.


Linux:
Older Redhat systems may have to use FC=f77 or FC=/usr/bin/f77
gcc  -m32 does not work with gfortran on gcc version 4.1.2 20080704 (Red Hat 4.1.2-46)
gcc   -O and -02 do not work under gcc version 4.1.2 20080704 (Red Hat 4.1.2-46)


Suns:
FC=f77
CC=cc
Remember to use gmake rather than Sun's make.
In make.sioseis use  magsun.o rather than  magmacosx.o

See http://sioseis.ucsd.edu


CHANGES
               SIOSEIS updates

16)  1 Dec 88
	a) poutex - changed to use pointers (setptr)
	b) trplot - bad interpolation
	c) diex - host floating point and ap simulator
	d) velaed - VTYPE needed upcase
	e) stack - increased buffer size
17)  8 Dec
	a) doed - remove IF( ierror .NE. 0 ) RETURN
18)  12 Dec
	a) diex - fno/lno noinc didn't work
	b) diex - multiple fno/lno lists didn't work
	c) doex - fk polaru didn't work
	d) died - GOTO 130 if "No such parameter:
	e) doed - GOTO 130 if "No such parameter:
19)  ?? Dec
	a)   FK stuff
	b) contro, getpro - for process tx2tp
20) 2 Jan 89
	a) died, diex - added SSC stuff
21) 5 Jan
	a) inedit - fno 99999 didn't work
22) 25 Jan
	a) ploted - changed check for vscale to be 200*1.25
	b) getoke - quotes didn't work anymore!
	c) getoke - added # or ! or { to be line terminators (allow comments)
23) 1 Feb
	a) segded, segdex, seg20 - convert to VMS
	b) gather - convert to VMS
	c) diex - si not working
	d) trplot - fix the bad interpolation change of 1 Dec.
24) 6 Feb
	a) ploted - remove vscale check altogether
25) 8 Feb
	a) stack - not big enough buffer for big delays
26) 10 Feb
	a) diex - renum on forgat didn't work
27) 11 Feb
	a) muteex - spatial variation on UNIX didn't work - f77 bug
28) 17 Feb
	a) ploted - comments
	b) died, diex - save segy headers as EBCDIC
	c) tx2fex, fk2tex, mrgfk, splitf, diex, doex - bad segy binary
	   header length
29) ?? Apr
	a) ibm2fp - very small numbers (from the Cray) were wrong
	b) diex - ntrgat not honored
30) 12 Apr
	a) stack - changed check for dead trace to GT 1 rather than
	   NE 1 - Moore's data had live trace = 0
	b) inputx - fis 99999 ignored ftr and ltr
	c) rdtrc, wrttrc - bad unit number for internal segy headers
31) 13 Apr
	a) stack - was renumbering the output!
	b) diex - calls to mrgfk used iptype rather than idtype
	    "bad "sort" from binary header" huh?
32) 14 Apr
	a) wrttrc - made the calls to xx2ibm machine independent
	b) stack, setptr - header word 17 = stack fold, 51 = -1
33) 17 Apr
	a) rdtrc - changed logic on shots with trace number 0
        b) decoed, decoex, decon, dconv - added parameter double - NOT TESTED
34) 27 Apr
	a) diex - fgmt/lgmt didn't work
35) 1 May
	a) fkmiex - nx not set properly (dx was wrong because of that)
36) 3 May
	a) ploted, plotex - added parameter plotsi (not completed)
37) 5 May
	a) rdtrc - changed rewind from 10 to 30 for VMS tape I/O bug (qad)
38) 8 May
	a) tx2fex - changed inap call - length after power of 2!
39) 11 May
	a) shifts.c - Alliant needs first arg to be short rather than int
40) 16 May
	a) diex - changed 31.b back
	b) mrgfk - added if( ftr and ltr not given and shot data
41) 24 May
	a) gathed - restart (frp) never converted to version 2 input lists
42) 25 May
	a) diskio.c - frefil closed and deleted ALL files, including stdin, stdout
43) 1 June
	a) segdex - stime and secs given and stime > delay and delay not 0
	  caused bad number of samples to be calculated
	b) segdex - ffilen and lfilen were not working
44) 12 June - Changes Alistair made to fdmigr
	a)  contro and diskio.f - Cray only - Change for Alistair's LINK
	b)  fdmiex.f, fdmvel.f, fdmlin.f, fdmied
	c)  add entries timer and second to dummies.f
	d) remove vapsim.f, avintr.f, avbufi.f
	e)  add myintr.f, mybufi.f, myspin.f, xslice.f
45) 15 June
	a)  fdmied - remove maxdip
	b)  diskio.c - increase max units to 30
46) 19 June
	a)  diex - SSC trace headers are now 1024 bytes.
47) 26 June
	a)  udecex - prewhittener on nonap was wrong
48) 28 June
	a)  segdex - more stime/secs bugs when delay NE 0
	b)  magtap.sun - remove some print statements
        c)  rdtrc - tape changes when different unit numbers didn't work
49) 10 July
	a)  diex - had format ssc honor forgat 1
	b)  diex - got forgat 1 to honor noinc
50) 17 July
	a)  dcode - was modifying nchars argument (didn't bother SIOSEIS!)
51) 19 July
        a)  muteex - print a warning when mute kills a trace
        b)  died - correct documentation
        c)  doed - correct documentation (add posaft)
52) 20 July
	a)  filtex - fixed and redid the spatial variation stuff
	b)  filted - changed the documentation about spatial variation
53) 24 July
	a)  decoex - fixed and redid the spatial variation stuff.
   ***  THIS MUST INCLUDE THE 17 APR. DECON UPDATE FOR DOUBLE  ***
	b)  diex - fkpolaru didn't work right (dskpos and idtype probs.)
	b)  mrgfk - fkpolaru didn't work right (dskpos=1 on non apollo)
54) 25 July
	a)  filted - set the filter length counter ntotal =0 after each list
55) 26 July
	a)  wrttrc - same as April! - bad header unit number in common readt
56)  9 August
	a)  contro - increase maximum trace length to 30k samples (fkmigr
	             uses 3 buffers worth! so max fkmigr is 8k)
	b)  contro - put maxsamps into common /edits/
57)  11 August
	a)  fdmied - changed the documentation for gamma.
58)  16 August
	a)  diex - removed 2 useless lines for SSC and trace headers
	b)  velaex - put the constant velocity value in SEGY header ibuf(46)
        c)  setptr - added icvelptr
59)  18 August
	a)  smuted, smutex - add parameter interp.
	b)  nmoapp - zero the data if before the start (for move in)
60)  23 August
	a)  gentl, trplot - change dtable for Vax (Unix Vax only?)
	b)  filtex - bug in the July 20 changes!
61)  25 August
	a)  inedit, inputx - added forgat > 1 for combining rps
	b)  tx2ted, tx2tex - added ohpath for using tx headers
62)  28-30 August
	a)  diex - added forgat > 1 for combining rps
	b)  diex - added nsamps and isi to binary header when SSC
	c)  diex - extended an all zero trace header to mean EOF
63)  4 Sept.
	a)  wrttrc - Allow a trace id of 0 when positioning after.
	b)  magtap.sun - give good message when > 32768 bytes are written
64)  19 Sept.
	a)  diex - Honor fsec and lsec
	b)  rdtrc - put isi, and nsamps into the SEGY binary header when
	SSC input so that the VISTA program works
65)  20 Sept.
	a)  wrttrc - VMS couldn't write IBM tapes (filter or not)
	b)  wrttrc - no machine could write IBM tapes when filter in procs

lsd.f
C    A PROGRAM TO READ AND LIST A SEGY FORMATTED DISK
c  -K = list Knudsen header items.
c  -O = Odec - NOT IMPLEMENTED
c
c  mod 5 Feb 14 - honor segy rev 1 delay scalar
c  mod ??????   - honor segy rev 1 water bottom scalar
c  mod 28 Oct 14 - Replace iargc with command_argument_count()
c  mod 28 Oct 14 - Replace getarg with command_argument()
c  mod 24 Mar 14 - Finish some never finished changes on coordinates
c
      INTEGER*2 ibuf
      CHARACTER*200 name, token
      REAL*8 dtemp
      DIMENSION BUF(70000),IBUF(222),LBUF(222)
      EQUIVALENCE (BUF(1),IBUF(1)),(BUF(1),LBUF(1))
      DATA icompt/7/, ndone/0/, long/0/, ntodo/999999/
      LOGICAL knudsen/.FALSE./, odec/.FALSE./


      nargs = command_argument_count() + 1
      numarg = 1
      CALL get_command_argument( numarg, name, length, istat)
      IF( name(1:1) .EQ. '-' ) THEN
          IF( name(2:2) .EQ. 'K' .OR. name(2:2) .EQ. 'k' )knudsen=.TRUE.
          IF( name(2:2) .EQ. 'O' ) odec = .TRUE.
          CALL get_command_argument( numarg, name, length, istat )
          numarg = numarg + 1
      ENDIF
      IF( nargs .LT. 2 .OR. nargs .GT. 6 ) THEN
          PRINT *,' lsd (list SEG-Y disk file) usage:',
     &      ' lsd [-K] SEGY-filename [trace-inc ntodo long-list]'
          STOP
      ENDIF
      CALL getfil( 4, lun, name, istat )
      IF( istat .NE. 0 ) THEN
          PRINT *,' Could not open file:   ',name(1:50)
          STOP
      ENDIF
      numarg = numarg + 1
      IF( is_big_endian() .LT. 0 ) icompt = 4
      nskip = 1
      IF( nargs .GT. numarg ) THEN
          CALL get_command_argument( numarg, token, length, istat)
          numarg = numarg + 1
          READ( token, * ) nskip
          IF( nargs .GT. numarg ) THEN
              CALL get_command_argument( numarg, token, length, istat)
              numarg = numarg + 1
              READ( token, * ) ntodo
          ENDIF
          IF( nargs .GT. numarg ) THEN
              CALL get_command_argument( numarg, token, length, istat)
              numarg = numarg + 1
              long = 1
          ENDIF
      ENDIF
      IF( nskip .GT. 0 ) NSKIP=NSKIP-1
      CALL rddisc( lun, buf, 800, istat )
      IF( istat .NE. 800 ) THEN
          PRINT *,' ***  ERROR  ***  disk file incorrect.',
     *        ' wanted 800 words, read ',istat,' words.'
          STOP
      ENDIF
      CALL rddisc( lun, buf, 100, istat )
      IF( istat .NE. 100 ) THEN
          PRINT *,' ***  ERROR  ***  disk file incorrect.',
     *            ' wanted 100 words, read ',istat,' words.'
          STOP
      ENDIF
      IF( icompt .EQ. 2 .OR. icompt .EQ. 4 ) CALL swap16( ibuf, 200 )
      idtype = ibuf(13)
      segyrev = REAL(ibuf(151)) / 256.
      ltemp = ibuf(153) * 3200
      IF( ibuf(151) .GT. 256 .AND. ibuf(151) .LT. 512 .AND.
     &     ibuf(153) .GT. 0 ) CALL podiscb( lun, 2, ltemp )
c****  This will fail if the number of headers is unknown (ibuf(153) = -1)
      PRINT 40
40    FORMAT('      SHOT   TR       RP    TR ID  RANGE ',
     *   '  DELAY NSAMPS    SI   YR DAY HR MIN SEC')
50    FORMAT(I10,I6,I8,I6,I3,I7,F8.2,I7,I6,I5,I4,I3,I4,I4)
  110 CONTINUE
      CALL rddisc( lun, buf, 60, istat)
      IF( istat .EQ. -1 ) THEN
          PRINT *,' Read End-Of-File.'
          STOP
      ENDIF
      IF( istat .NE. 60 ) THEN
          PRINT *,' ***  ERROR  ***  disk file incorrect.',
     *            ' wanted 60 words, read ',istat,' words.'
          STOP
      ENDIF
      IF( icompt .EQ. 2 .OR. icompt .EQ. 4 ) THEN
          CALL swap32( lbuf, 7 )
          CALL swap16( ibuf(15), 4 )
          CALL swap32( lbuf(10), 8 )
          CALL swap16( ibuf(35), 2 )
          CALL swap32( lbuf(19), 4 )
          CALL swap16( ibuf(45), 40 )  ! 45-84, bytes 90-168
          CALL swap16( ibuf(85), 35 )  ! rev 1 did away with the sioseis use of "unassigned"
c          IF( .NOT. knudsen ) THEN
c              CALL swap32( lbuf(46), 8 )
c              CALL swap16( ibuf(110), 1 )
c          ENDIF
      ENDIF
      CALL ushort2long( ibuf(58), nsamps )
      delay = ibuf(55)
      IF( ibuf(108) .NE. 0 .AND. segyrev .GE. 1. ) THEN    ! rev 1 delay scalar
          IF( ibuf(108) .GT. 0 ) THEN
              delay = delay * ibuf(108)
          ELSE
              delay = delay / (-1 * ibuf(108) )
         ENDIF
      ENDIF
c***  a negative delay is possible
      PRINT 50,LBUF(3),LBUF(4),LBUF(6),LBUF(7),IBUF(15),LBUF(10),
     *        delay,nsamps,IBUF(59),(IBUF(I),I=79,83)
      IF( long .NE. 0 ) THEN
          relev = REAL(lbuf(11))
          selev = REAL(lbuf(12))
          sdep = REAL(lbuf(13))
          wdep = REAL(lbuf(16))
          IF( ibuf(35) .GT. 0 ) THEN
              relev = relev * REAL(ibuf(35))
              selev = selev * REAL(ibuf(35))
              sdep = sdep * REAL(ibuf(35))
              wdep = wdep * REAL(ibuf(35))
          ENDIF
          IF( ibuf(35) .LT. 0 ) THEN
              relev = -relev / REAL(ibuf(35))
              selev = -selev / REAL(ibuf(35))
              sdep = -sdep / REAL(ibuf(35))
              wdep = -wdep / REAL(ibuf(35))
          ENDIF
          PRINT *,' shot',lbuf(3),' tr',lbuf(4),' src',lbuf(5),
     &        ' rp',lbuf(6),' tr',lbuf(7),' id',ibuf(15),' eog',lbuf(51)
          PRINT *,' fold',ibuf(17),' range',lbuf(10),' relev',relev,
     &        ' selev',selev,' sdep',sdep,' wdep',wdep
          PRINT *,' time switch or shot time millisecond ',ibuf(84)
c****   ibuf(45) = 2, then seconds of arc
c****            = 3, then decimal degrees
c****            = 4, then +/-DDDMMSS.ss
          IF( lbuf(19)+lbuf(20) .NE. 0 ) THEN
              IF( ibuf(45) .EQ. 2 ) THEN
                PRINT *,' Coordinates in seconds of arc and had scalar',
     &                    ibuf(36)
                  dtemp = DABS(DFLOAT(lbuf(19)))
                  IF( ibuf(36) .GT. 0 ) dtemp = dtemp * DFLOAT(ibuf(36))
                  IF(ibuf(36) .LT. 0) dtemp=DABS(dtemp/DFLOAT(ibuf(36)))
                  temp = dtemp
                  CALL secsdms( 1, temp, ideg, min, sec )
                  IF( lbuf(19) .LT. 0 ) ideg = -ideg
                  PRINT *,' source long ',ideg, min, sec,
     &                    '(',temp,'arcsecs)'
                  dtemp = DABS(DFLOAT(lbuf(20)))
                  IF( ibuf(36) .GT. 0 ) dtemp = dtemp * DFLOAT(ibuf(36))
                  IF(ibuf(36) .LT. 0) dtemp=DABS(dtemp/DFLOAT(ibuf(36)))
                  temp = dtemp
                  CALL secsdms( 1, temp, ideg, min, sec )
                  IF( lbuf(20) .LT. 0 ) ideg = -ideg
                  PRINT *,' source lat. ',ideg, min, sec,
     &                    '(',temp,'arcsecs)'
              ELSEIF( ibuf(45) .EQ. 3 .OR. ibuf(45) .EQ. 5 ) THEN
c****   3 means INTEGER decimal degrees, but I haven't seen that, but ODEC has floats
c****   5 is what I want to see for FLOAT decimal degrees
               PRINT *,' Coordinates in decimal degrees and had scalar',
     &                    ibuf(36)
                  PRINT *,' Source and receiver xy coordinates:',
     &                  buf(19), buf(20), buf(21), buf(22)
                  dtemp = (ABS(buf(19)))
                  IF( ibuf(36) .GT. 0 ) dtemp = dtemp * DFLOAT(ibuf(36))
                  IF(ibuf(36) .LT. 0) dtemp=DABS(dtemp/DFLOAT(ibuf(36)))
                  temp = dtemp * 3600.
                  CALL secsdms( 1, temp, ideg, min, sec )
                  IF( buf(19) .LT. 0 ) ideg = -ideg
                  PRINT *,' source long. ',ideg, min, sec,
     &                '(',temp,'arcsecs)'
                  dtemp = (ABS(buf(20)))
                  IF( ibuf(36) .GT. 0 ) dtemp = dtemp * DFLOAT(ibuf(36))
                  IF(ibuf(36) .LT. 0) dtemp=DABS(dtemp/DFLOAT(ibuf(36)))
                  temp = dtemp * 3600.
                  CALL secsdms( 1, temp, ideg, min, sec )
                  IF( buf(20) .LT. 0 ) ideg = -ideg
                  PRINT *,' source lat. ',ideg, min, sec,
     &                '(',temp,'arcsecs)'
              ELSEIF( ibuf(45) .EQ. 1 ) THEN
                  PRINT *,' Coordinates in length and had scalar',
     &                    ibuf(36)
                  dtemp19 = DABS(DFLOAT(lbuf(19)))
                  IF( ibuf(36) .GT. 0 )
     &                dtemp19 = dtemp19 * DFLOAT(ibuf(36))
                  IF(ibuf(36) .LT. 0)
     &                dtemp19=DABS(dtemp19/DFLOAT(ibuf(36)))
                  dtemp20 = DABS(DFLOAT(lbuf(20)))
                  IF( ibuf(36) .GT. 0 )
     &                dtemp20 = dtemp20 * DFLOAT(ibuf(36))
                  IF(ibuf(36) .LT. 0)
     &                dtemp20=DABS(dtemp20/DFLOAT(ibuf(36)))
                  PRINT *,' Source coordinates: ', dtemp19, dtemp20
                  dtemp21 = DABS(DFLOAT(lbuf(21)))
                  IF( ibuf(36) .GT. 0 )
     &                dtemp21 = dtemp21 * DFLOAT(ibuf(36))
                  IF(ibuf(36) .LT. 0)
     &                dtemp21=DABS(dtemp21/DFLOAT(ibuf(36)))
                  dtemp22 = DABS(DFLOAT(lbuf(22)))
                  IF( ibuf(36) .GT. 0 )
     &                dtemp22 = dtemp22 * DFLOAT(ibuf(36))
                  IF(ibuf(36) .LT. 0)
     &                dtemp22=DABS(dtemp22/DFLOAT(ibuf(36)))
                  PRINT *,' Receiver coordinates: ', dtemp21, dtemp22
              ELSE
                  PRINT *,' Coordinate units:',ibuf(45),' scalar:',
     &               ibuf(36)
                  PRINT *,' Source coordinates: ', lbuf(19), lbuf(20)
                  PRINT *,' Receiver coordinates: ', lbuf(21), lbuf(22)
              ENDIF
          ENDIF
          IF( ibuf(64)+ibuf(65) .NE. 0 ) THEN
              PRINT *,' Correlation switch (1=no, 2=yes):', ibuf(63)
              PRINT *,' Sweep frequency at start (Hz).', ibuf(64)
              PRINT *,' Sweep frequency at end (Hz).', ibuf(65)
          ENDIF
          IF( ibuf(66) .NE. 0 ) PRINT *,' Sweep length:',ibuf(66)
          IF( knudsen ) THEN
c****         Valid on raw files only.  sioseis diskox doesn't swap the same way.
              IF( icompt .EQ. 2 .OR. icompt .EQ. 4 )
     &            CALL swap16( ibuf(95), 20 )
              PRINT *,' PingStartTimeMS:',ibuf(95)
              PRINT *,' TxPower:',ibuf(96)
              PRINT *,' RxGain:',ibuf(97)
              PRINT *,' ProcessingGain:',ibuf(98)
              PRINT *,' Sensitivity:',ibuf(99)
              PRINT *,' MuxChannel:',ibuf(100)
              PRINT *,' EchoStrength:',ibuf(101)
              PRINT *,' PrimaryChannel:',ibuf(102)
              PRINT *,' Kel_PulseLength parameter:',ibuf(103)
              PRINT *,' TxBlank:',ibuf(104)
              PRINT *,' SoundSpeed:',ibuf(105)
              PRINT *,' StartDepth:',ibuf(106)
              PRINT *,' EndDepth:',ibuf(107)
              PRINT *,' EchoThreshold:',ibuf(108)
              PRINT *,' Heave:',ibuf(109)
              PRINT *,' HeaveLatency:',ibuf(110)
              PRINT *,' GpsLatency:',ibuf(111)
              PRINT *,' EventMarkCode:',ibuf(112)
              PRINT *,' EventMarkNumber:',ibuf(113)
              PRINT *,' Scaler:',ibuf(114)
              IF( icompt .EQ. 2 .OR. icompt .EQ. 4 )
     &            CALL swap32( lbuf(58), 1 )
              IF( ibuf(114) .LT. 0 )
     &            temp = - FLOAT(lbuf(58)) / FLOAT(ibuf(114))
              PRINT *,' DataRate:',temp
          ENDIF
      ENDIF
      ndone = ndone + 1
      IF( ndone .GE. ntodo ) STOP
      IF( idtype .EQ. 3 .OR. idtype .EQ. 4 ) THEN
          nbytes = nsamps * 2
      ELSE
          nbytes = nsamps * 4
      ENDIF
      IF( nsamps .GT. 0 ) THEN
          CALL rddiscb( lun, buf, nbytes, istat )
          IF( istat .NE. nbytes ) THEN
              PRINT *,' ***  ERROR  ***  disk file incorrect.',
     *            ' wanted',nbytes,' bytes, read ',istat,' bytes.'
              STOP
          ENDIF
      ENDIF
      IF( NSKIP .GT. 0 ) THEN
          DO 150 I=1,NSKIP
              CALL rddisc( lun, buf, 60, istat)
              IF( istat .EQ. -1 ) STOP
              IF( istat .NE. 60 ) THEN
                  PRINT *,' ***  ERROR  ***  disk file incorrect.',
     *            ' wanted 60 words, read ',istat,' words.'
                  STOP
              ENDIF
              IF(icompt.EQ.2.OR.icompt.EQ.4) CALL swap16(ibuf(58),1)
              CALL ushort2long( ibuf(58), nsamps )
              IF( ibuf(15) .EQ. 28 .AND. nsamps .NE. 1024 )
     &           nsamps = (2048 - 240) / 4
              IF( idtype .EQ. 3 .OR. idtype .EQ. 4 ) THEN
                  nbytes = nsamps * 2
              ELSE
                  nbytes = nsamps * 4
              ENDIF
              IF( nsamps .GT. 0 ) THEN
                  CALL rddiscb( lun, buf, nbytes, istat )
                  IF( istat .NE. nbytes ) THEN
                      PRINT *,' ***  ERROR  ***  disk file incorrect.',
     *                  ' wanted',nbytes,' bytes, read ',istat,' bytes.'
                      STOP
                  ENDIF
              ENDIF
  150     CONTINUE
      ENDIF
      GO TO 110
      END
